jerry-core/lit/lit-magic-strings.c

```diff
@@ -228,7 +228,7 @@ lit_is_utf8_string_pair_magic (const lit_utf8_byte_t *string1_p, /**< first utf-
   while (first <= last)
   {
     int middle = ((first + last) / 2); /**< mid point of search */
-    const lit_utf8_byte_t *middle_string_p = lit_get_magic_string_utf8 (middle);
+    const lit_utf8_byte_t *middle_string_p = lit_get_magic_string_utf8 ((lit_magic_string_id_t) middle);
 
```

Comment:
Discussed this offline. Conclusion: A better approach is to rewrite the binary search algorithm to use unsigned integers rather than signed integers for the indices, this avoids the need add a cast for the `lit_get_magic_string_utf8()` call.